botstrap.py

```diff
@@ -147,37 +219,82 @@ def get_all_channels_and_categories(self) -> tuple[dict[str, str], dict[str, str
                 name = f"off_topic_{off_topic_count}"
                 off_topic_count += 1
 
-            if channel_type == 4:
+            if channel_type == GUILD_CATEGORY_TYPE:
                 categories[name] = channel["id"]
             else:
                 channels[name] = channel["id"]
 
         return channels, categories
 
-    def webhook_exists(self, webhook_id_: int) -> bool:
-        """A predicate that indicates whether a webhook exists already or not."""
-        try:
-            self.get(f"webhooks/{webhook_id_}")
-            return True
-        except HTTPStatusError:
-            return False
+    def get_all_guild_webhooks(self) -> list[dict[str, Any]]:
+        """Lists all the webhooks for the guild."""
+        response = self.get(f"/guilds/{self.guild_id}/webhooks")
+        return response.json()
 
     def create_webhook(self, name: str, channel_id_: int) -> str:
         """Creates a new webhook for a particular channel."""
         payload = {"name": name}
 
-        response = self.post(f"channels/{channel_id_}/webhooks", json=payload)
+        response = self.post(
+            f"/channels/{channel_id_}/webhooks",
+            json=payload,
+            headers={"X-Audit-Log-Reason": "Creating webhook as part of PyDis botstrap"},
+        )
         new_webhook = response.json()
         return new_webhook["id"]
 
+    def list_emojis(self) -> list[dict[str, Any]]:
+        """Lists all the emojis for the guild."""
+        response = self.get(f"/guilds/{self.guild_id}/emojis")
+        return response.json()
+
+    def get_emoji_contents(self, emoji_id: str | int) -> bytes | None:
+        """Fetches the image data for an emoji by ID."""
+        # emojis are located at https://cdn.discordapp.com/emojis/821611231663751168.png?size=4096
```

Comment:
I'd use path-params to document this and where that snowflake comes from (even though it's obvious). Also, not sure the size is necessary given we don't then use it for the subsequent request.
```suggestion
        # Emojis are located at https://cdn.discordapp.com/emojis/{emoji_id}.png
```